dash.el

```diff
@@ -2274,9 +2274,22 @@ The test for equality is done with `equal',
 or with `-compare-fn' if that's non-nil.
 
 Alias: `-uniq'"
-  (let (result)
-    (--each list (unless (-contains? result it) (!cons it result)))
-    (nreverse result)))
+  ;; Implementation note: Using a hash-table to remove duplicates is
+  ;; considerably faster than do it by iteration when the length
+  ;; of the list is greater than 32.
+  (let* ((-compare-fn (or -compare-fn #'equal))
```

Comment:
I think it's cleaner not to bind `-compare-fn` all over the place. The analysis and manipulation of its value should be as focussed as possible. In this case, we need only test its value for the purpose of allocating the lookup table:

```el
(let* ((len (length list))
       (lut (and (> len 32)
                 (memq -compare-fn '(nil equal eq eql))
                 (make-hash-table :test (or -compare-fn #'equal) :size len))))
```